src/main/java/tornadofx/ItemControls.kt

```diff
@@ -261,14 +261,12 @@ class LazyTreeItem<T : Any>(
         val itemProcessor: (LazyTreeItem<T>) -> Unit = {},
         val childFactory: (TreeItem<T>) -> List<T>?
 ) : TreeItem<T>(value) {
-    var leafResult: Boolean? = null
+    private val leafResult: Boolean by lazy { leafCheck(this) }
     var childFactoryInvoked = false
     var childFactoryResult: List<T>? = null
 
     override fun isLeaf(): Boolean {
-        if (leafResult == null)
-            leafResult = leafCheck(this)
-        return leafResult!!
+        return leafResult
```

Comment:
remove the body: `override fun isLeaf() = leafResult`

Reply:
Sounds reasonable. Should I fill new PR with only this one change?
There are other places where some small refactorings could be done, so maybe this can be left as is right now.